### PR TITLE
Exception for shared mem

### PR DESCRIPTION
**Pull request by SirTobias**
# Issue https://github.com/Project-OSRM/osrm-backend/issues/5998

Give reasonable messages for the exceptions of Boost that at least some messages pop up if the shared memory directory cannot be created or initialized.

## Tasklist

 - [x ] CHANGELOG.md entry ([How to write a changelog entry](http://keepachangelog.com/en/1.0.0/#how))
 - [-] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki) -> imho not relevant
 - [- ] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md) - imho not relevant
 - [ ] review
 - [ ] adjust for comments
 - [ ] cherry pick to release branch

### CODE REVIEW

File: include/storage/shared_monitor.hpp [hunk 1]
```diff
@@ -42,7 +42,16 @@ template <typename Data> struct SharedMonitor
...
     {
-        shmem = bi::shared_memory_object(bi::open_or_create, Data::name, bi::read_write);
+        try
+        {
+            shmem = bi::shared_memory_object(bi::open_or_create, Data::name, bi::read_write);
+        }
+        catch (const bi::interprocess_exception &exception)
+        {
+            auto message =
+                boost::format("could not create directory or file '%1%' \n") % exception.what();
```
Comment (mjjbell) [inline include/storage/shared_monitor.hpp:52]: I would include the name of the memory block to make it consistent with the other exception messages below.
```suggestion
                boost::format("Unable to open or create shared memory block '%1%'") % (const char *)Data::name;
```

File: include/storage/shared_monitor.hpp [hunk 2]
```diff
@@ -126,7 +135,9 @@ template <typename Data> struct SharedMonitor
         }
         catch (const bi::interprocess_exception &exception)
         {
-            return false;
+            auto message =
```
Comment (mjjbell) [inline include/storage/shared_monitor.hpp:138]: This changes the expected behaviour of this function. The exception is an indication of the shared memory object not existing.

**Review by mjjbell — CHANGES_REQUESTED** [review]

**Review by SirTobias — COMMENTED** [review]
Dear Mr. Bell, thank you for the comments. I revised my changes according to your requests.

File: include/storage/shared_monitor.hpp [hunk 3]
```diff
@@ -42,7 +42,16 @@ template <typename Data> struct SharedMonitor
 
     SharedMonitor(const Data &initial_data)
     {
-        shmem = bi::shared_memory_object(bi::open_or_create, Data::name, bi::read_write);
+        try
+        {
+            shmem = bi::shared_memory_object(bi::open_or_create, Data::name, bi::read_write);
```
Comment (SiarheiFedartsou) [inline include/storage/shared_monitor.hpp:47]: @mjjbell I saw you had concerns about this PR. Is it okay now? I could probably rebase and merge it

File: include/storage/shared_monitor.hpp [hunk 4]
```diff
@@ -126,6 +135,8 @@ template <typename Data> struct SharedMonitor
         }
         catch (const bi::interprocess_exception &exception)
         {
+            auto message = boost::format("could not create for shared memory block '%1' the directory or file '%2%' \n") %
```
Comment (SiarheiFedartsou) [inline include/storage/shared_monitor.hpp:138]: Seems to be unused variable

File: include/storage/shared_monitor.hpp (hunk 3 above)
Comment (mjjbell) [reply include/storage/shared_monitor.hpp:47]: I can't remember exactly. I think it looked unfinished.

**Comment by DennisOSRM** [comment]
closing stale PR. Reopen if still relevant.

**Pull request closed without merging**